app/app.go

```diff
@@ -714,3 +761,12 @@ func (a *app) SetDeviceStateIntegration(
 func (a *app) GetEvents(ctx context.Context, filter model.EventsFilter) ([]model.Event, error) {
 	return a.store.GetEvents(ctx, filter)
 }
+
+func runAndLogError(ctx context.Context, f func() error) {
+	err := f()
+	if err != nil {
+		log.FromContext(ctx).
+			WithField("panic", err.Error()).
+			Error("failed to process an asynchronous webhook")
+	}
```

Comment:
On the other hand, it would be useful to catch panics and not let it bring down the application...
```suggestion
	var err error
	logger := log.FromContext(ctx)
	defer func() {
		if r := recover(); r != nil {
			logger.WithField("panic", r).
				Error("panic processing asynchronous webhook")
		} else if err != nil {
			logger.WithField("error", err.Error()).
				Error("failed to process an asynchronous webhook")
		}
	}()
	err = f()
```